### PR TITLE
Document correlation between credential iterations and query latency

### PR DESCRIPTION
**Pull request by techdocsmith**
### Description

Documents the correlation between credential iterations and query latency. Includes some minor additional stylistic changes.

This PR has:
- [x] added documentation for features or behaviors.

### CODE REVIEW

**Comment by techdocsmith** [comment]
@suneet-s , @2bethere PTAL.

File: docs/development/extensions-core/druid-basic-security.md [hunk 1]
```diff
@@ -62,16 +69,16 @@ druid.auth.authenticator.MyBasicMetadataAuthenticator.authorizerName=MyBasicMeta
 ```
 
 To use the Basic authenticator, add an authenticator with type `basic` to the authenticatorChain.
-The authenticator needs to also define a credentialsValidator with type 'metadata' or 'ldap'.
-If credentialsValidator is not specified, type 'metadata' will be used as default.
+The authenticator must define a credentials validator `credentialsValidator` with type 'metadata' or 'ldap'.
+Otherwise the credentials validator defaults to `metadata` if you do not specify another type.
```
Comment (nishantmonu51) [inline docs/development/extensions-core/druid-basic-security.md:73]: above 2 lines seems to be contradictory, first one says you must define it and 2nd mentions a default value. 

probably, reword as - "The authenticator uses metadata credentials validator by default which can be configured to `metadata`." Or similar wordings.

**Review by nishantmonu51 — APPROVED** [review]
LGTM

File: docs/development/extensions-core/druid-basic-security.md [hunk 2]
```diff
@@ -81,11 +88,17 @@ The authenticator configuration examples in the rest of this document will use "
...
 |`druid.auth.authenticator.MyBasicMetadataAuthenticator.cacheNotificationTimeout`|The timeout in milliseconds for the cache notifications.|5000|No|
-|`druid.auth.authenticator.MyBasicMetadataAuthenticator.credentialIterations`|Number of iterations to use for password hashing.|10000|No|
+|`druid.auth.authenticator.MyBasicMetadataAuthenticator.credentialIterations`|Number of iterations to use for password hashing. See [Credential iterations and query times](#credential-iterations-and-query-times)|10000|No|
 |`druid.auth.authenticator.MyBasicMetadataAuthenticator.credentialsValidator.type`|The type of credentials store (metadata) to validate requests credentials.|metadata|No|
 |`druid.auth.authenticator.MyBasicMetadataAuthenticator.skipOnFailure`|If true and the request credential doesn't exists or isn't fully configured in the credentials store, the request will proceed to next Authenticator in the chain.|false|No|
 |`druid.auth.authenticator.MyBasicMetadataAuthenticator.authorizerName`|Authorizer that requests should be directed to|N/A|Yes|
 
+##### Credential iterations and query times
```
Comment (suneet-s) [inline docs/development/extensions-core/druid-basic-security.md:96]: The iterations apply to all API calls, this title makes me think it only impacts query times

**Review by suneet-s — COMMENTED** [review]
+1 on @nishantmonu51's suggestion

File: docs/development/extensions-core/druid-basic-security.md (hunk 2 above)
Comment (techdocsmith) [reply docs/development/extensions-core/druid-basic-security.md]: @suneet-s I have updated this to reflect that the impact is on all API calls, but in the first sentence mention "including queries"

File: docs/development/extensions-core/druid-basic-security.md (hunk 1 above)
Comment (techdocsmith) [reply docs/development/extensions-core/druid-basic-security.md]: @nishantmonu51 , @suneet-s updated this to reflect that `metadata` is the default. And you only need to define one for `LDAP`. PTAL. Not sure if I've 100% got the nuance here.

**Review by suneet-s — APPROVED** [review]
Thanks @techdocsmith ! LGTM

**Pull request merged**